src/Sentinel.php

```diff
@@ -309,7 +309,7 @@ public function authenticate($credentials, bool $remember = false, bool $login =
 
             $valid = $user !== null ? $this->users->validateCredentials($user, $credentials) : false;
 
-            if ($user === null || ! $valid) {
+            if (! $valid) {
```

Comment:
The check seems different now i suppose, since above it checks user is not null, here it was checking user was null and not valid.

Reply:
$user can only be null or ```EloquentUser```. If user is null, valid will always be false because the user is null.
If the comparison was ```&&``` instead of ``||``` I would understand keeping the null check but in the current state $valid can never be true if $user is null.

Reply:
Sorry, i misread the code when i commented out.

It makes sense.